src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyQueueCapacityAction.java

```diff
@@ -56,6 +57,7 @@ private ModifyQueueCapacityAction(NodeKey esNode, ResourceEnum threadPool, AppCo
     this.lowerBound = lowerBound;
     this.upperBound = upperBound;
     this.canUpdate = canUpdate;
+    persistValues();
```

Comment:
Why do we want to persist values as soon as an action is created? Shouldn't we just persist them when an action is published, why would we log in db the actions that were created but discarded or pruned?

The `Persistable` interface which actions implement can have a `persistToDb()` fn that a `PublisherEventsPersistor implements ActionListener` calls. (can rename to better names)

Or better still, we can have the `ActionListener` hold all the logic for persisting to sql, and just call `Action` methods like `name(), impactedNodes(), summary(), coolOffPeriodInMillis(), isActionable()` to get values and persist them.